raiden/connection_manager.py

```diff
@@ -227,14 +228,28 @@ def retry_connect(self):
                         self.token_address,
                         partner
                     )
-                    self.api.deposit(
-                        self.token_address,
-                        partner,
-                        self.initial_funding_per_partner
-                    )
                 # this can fail because of a race condition, where the channel partner opens first
-                except Exception as e:
-                    log.error('could not open a channel', exc_info=e)
+                except DuplicatedChannel:
+                    log.error('partner opened channel first')
+
+                try:
+                    with gevent.timeout.Timeout(30):
```

Comment:
If I understood correctly, the `open` transaction was executed concurrently and the exception is raised in the losing node, correct? If that is the case, then one of the transaction must have been successful, and this must be known by the ethereum node, so there must be a new block with the `ChannelNew` event.

Instead of a loop here a `self.raiden.poll_blockchain_events()` must be sufficient, otherwise something else is going on.

Side note: This will potentially  fail badly with reorgs.